scripts/circle-ci-configuration.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+###############################################################################
+#
+#   circle-ci-configuration.sh
+#
+###############################################################################
+
+# From the sh(1) man page of FreeBSD:
+# Exit immediately if any untested command fails. in non-interactive
+# mode.  The exit status of a command is considered to be explicitly
+# tested if the command is part of the list used to control an if,
+# elif, while, or until; if the command is the left hand operand of
+# an “&&” or “||” operator; or if the command is a pipeline preceded
+# by the ! operator.  If a shell function is executed and its exit
+# status is explicitly tested, all commands of the function are con‐
+# sidered to be tested as well.
+set -e
+
+# Return status is that of the last command to fail in a
+# piped command, or a zero if they all succeed.
+set -o pipefail
+
+EXIT=0
+
+curl -sSL https://deb.nodesource.com/gpgkey/nodesource.gpg.key | sudo apt-key add -
+echo "deb https://deb.nodesource.com/node_6.x xenial main" | sudo tee /etc/apt/sources.list.d/nodesource.list
+echo "deb-src https://deb.nodesource.com/node_6.x xenial main" | sudo tee -a /etc/apt/sources.list.d/nodesource.list
+
+sudo apt-get update
+cat requirements/system/ubuntu/apt-packages.txt | DEBIAN_FRONTEND=noninteractive xargs apt-get -yq install
+service mongodb restart
```

Comment:
what do we need mongo for? I wonder, is this file taken from somewhere were they use Circle too to run the tests?

Mongo can be run as a service on circle: https://hub.docker.com/r/circleci/mongo/

Reply:
Mongo is used to store all the course content. It is being installed by `requirements/system/ubuntu/apt-packages.txt`. I was occasionally running into issues with mongo not responding after the install and a restart solves that so just doing it here. Since the default deb is working fine, it makes more sense to use it than to add additional configuration for the Circle service.

Reply:
hey! I like the idea of using what CircleCI has to offer. Here you can see how we did it on OCIM: https://github.com/open-craft/opencraft/blob/master/circle.yml#L13

There is also this step: https://github.com/open-craft/opencraft/blob/master/circle.yml#L46 to wait till Mongo DB is accessible.

@clemente thoughts on this?

Reply:
sounds good!